go/vt/vttablet/tabletmanager/rpc_backup.go

```diff
@@ -84,12 +83,7 @@ func (agent *ActionAgent) Backup(ctx context.Context, concurrency int, logger lo
 		}
 		originalType = tablet.Type
 		// update our type to BACKUP
-		if _, err := topotools.ChangeType(ctx, agent.TopoServer, tablet.Alias, topodatapb.TabletType_BACKUP, nil); err != nil {
-			return err
-		}
-
-		// let's update our internal state (stop query service and other things)
-		if err := agent.refreshTablet(ctx, "before backup"); err != nil {
+		if err := agent.ChangeType(ctx, topodatapb.TabletType_BACKUP); err != nil {
```

Comment:
This is a problem. We are calling an RPC within an RPC, so it's blocking on the actionMutex.

Reply:
Another approach would be to create a `changeTypeLocked` which is called when we are already holding the action lock.

Reply:
I think we were taking the lock in this block so that, in the case of drain-for-backup, we exclude all other RPC actions while the backup is happening. If that's true, we still need to hold the lock throughout the rest of this function, not just while changing type. The pattern we typically use in Vitess for this is to pull the meat of ChangeType() out into changeTypeLocked() and call the latter whenever we already hold the lock.

Reply:
Good point. I forgot that defer only gets executed at the end of the function, not at the end of the block.